example/contracts/Migrations.sol

```diff
@@ -0,0 +1,21 @@
+contract Migrations {
+  address public owner;
+  uint public last_completed_migration;
+
+  modifier restricted() {
+    if (msg.sender == owner) _
+  }
+
+  function Migrations() {
+    owner = msg.sender;
+  }
+
+  function setCompleted(uint completed) restricted {
+    last_completed_migration = completed;
+  }
+
+  function upgrade(address new_address) restricted {
+    Migrations upgraded = Migrations(new_address);
+    upgraded.setCompleted(last_completed_migration);
+  }
```

Comment:
I don't think this works the way you want it to. The default authentication scheme only allows `setCompleted` to be called by the `owner`, but the old `Migrations` is not going to be the owner. Granted your new code might be different, but seems weird to have to write a new `Migrations` contract with the assumption that the old one can call it.

I don't think this is necessary anyway, since it's a single `uint` that will need to be ported to the new `Migrations` contract, and the new contract can get it itself by calling `last_completed_migration`.